nuxeo-core/nuxeo-core-convert/src/main/java/org/nuxeo/ecm/core/convert/service/ConversionServiceImpl.java

```diff
@@ -219,6 +231,97 @@ public static String getCacheBasePath() {
     }
 
     @Override
+    public Blob convertBlobToPDF(Blob blob) throws IOException {
+        String mimetype = blob.getMimeType();
+        if (MimetypeRegistry.PDF_MIMETYPE.equals(mimetype)) {
+            return blob;
+        }
+        Blob result;
+        if (MediaType.TEXT_PLAIN.equals(mimetype)) {
+            result = convertBlobToMimeType(blob, MimetypeRegistry.PDF_MIMETYPE);
+        } else {
+            // Convert the blob to HTML
+            if (!MediaType.TEXT_HTML.equals(mimetype)) {
+                String filename = blob.getFilename();
+                blob = convertBlobToMimeType(blob, MediaType.TEXT_HTML);
+                blob.setFilename(filename);
+            }
+            // Replace the image URLs by absolute paths
+            Path tempDirectory = Framework.createTempDirectory("blobs");
+            DownloadService downloadService = Framework.getService(DownloadService.class);
+            blob = replaceURLsByAbsolutePaths(blob, tempDirectory, downloadService::resolveBlobFromDownloadUrl);
+            // Convert the blob to PDF
+            result = convertBlobToMimeType(blob, MimetypeRegistry.PDF_MIMETYPE);
+            org.apache.commons.io.FileUtils.deleteQuietly(tempDirectory.toFile());
+        }
+        if (result != null) {
+            adjustPDFBlobName(blob, result);
+        }
+        return result;
+    }
+
+    protected Blob convertBlobToMimeType(Blob blob, String destinationMimeType) {
+        BlobHolder bh = new SimpleBlobHolder(blob);
+        bh = convertToMimeType(destinationMimeType, bh, new HashMap<String, Serializable>());
+        if (bh == null) {
+            return null;
+        }
+        Blob result = bh.getBlob();
+        return result;
+    }
+
+    protected static void adjustPDFBlobName(Blob in, Blob out) {
+        String filename = in.getFilename();
+        if (StringUtils.isBlank(filename)) {
+            filename = "file_" + System.identityHashCode(in);
+        }
+        int pos = filename.lastIndexOf('.');
+        if (pos > 0) {
+            filename = filename.substring(0, pos);
+        }
+        out.setFilename(filename + PDF_EXTENSION);
+        out.setMimeType(MimetypeRegistry.PDF_MIMETYPE);
+    }
+
+    /**
+     * Replace the image URLs of an HTML blob by absolute local paths.
+     *
+     * @throws IOException
+     * @since 9.1
+     */
+    protected static Blob replaceURLsByAbsolutePaths(Blob blob, Path tempDirectory, Function<String, Blob> blobResolver)
+            throws IOException {
+        String initialBlobContent = blob.getString();
+        // Find images links in the blob
+        Pattern pattern = Pattern.compile("(src=\")(.*?)(\")");
```

Comment:
It would be nice to match also the attributes delimited with `'`, you can do it using the pattern `"(src=([\"']))(.*?)(\\2)"` but the `.group(2)` (below) will become group 3 and the backreference `$3` will become `$4`.